### PR TITLE
Improve reliability

### PR DESCRIPTION
**Pull request by Yoric**

### CODE REVIEW

**Review by Gnuxie — APPROVED** [review]
Nice, looks good!

Have a couple of questions though

File: src/lib.rs [hunk 1]
```diff
@@ -981,9 +1001,24 @@ pub async fn up(docker: &Docker, config: &Config) -> Result<(), Error> {
     .context("Failed to start Synapse")?;
 
     debug!("Synapse is now launched");
-    handle_user_registration(config)
-        .await
-        .context("Failed to setup users")?;
+    // Sometimes, Synapse fails to launch and user registration loops endlessly.
```
[the remark this answers is not in the document]
Comment (Yoric) [reply src/lib.rs:1004]: It backs off endlessly.

[the remark this answers is not in the document]
Comment (Gnuxie) [reply src/lib.rs:1004]: Ok, that's a shame

File: src/registration.rs [hunk 2]
```diff
@@ -302,13 +318,40 @@ pub async fn handle_user_registration(config: &crate::Config) -> Result<(), Erro
...
+                }
                 request.room_alias_name = Some(alias.as_ref());
-                if joined_rooms.iter().any(|joined| {
-                    matches!(joined.canonical_alias(), Some(joined_alias) if joined_alias.as_str() == alias)
-                }) {
-                    // Don't re-register the room.
-                    continue;
+                // If the alias is already taken, we may need to remove it.
+                let full_alias = format!("#{}:{}", alias, config.homeserver.server_name);
+                debug!("Attempting to register alias {}, this may require unregistering previous instances first.", full_alias);
```
[the remark this answers is not in the document]
Comment (Yoric) [reply src/registration.rs:330]: Added to the README.

**Pull request merged**